crates/aptos-faucet/src/main.rs

```diff
@@ -111,7 +112,7 @@ mod tests {
         tokio::task::spawn(async move { future.await });
 
         let service = Service::new(
-            format!("http://localhost:{}/", address.port()),
+            Url::parse(&format!("http://localhost:{}/", address.port())).unwrap(),
```

Comment:
Note, `.port` is super suss with the URL library, I suspect what we really want here `port_or_known_default` (assuming `address` is a `Url`).

Reply:
Address is actually a SocketAddr, so it'll always have a value.  This code and how it entangles intaking port changes is a little suss in general.